schema-engine-tag/src/main/java/org/apache/iotdb/db/metadata/tagSchemaRegion/tagIndex/flush/MemChunkFlush.java

```diff
@@ -0,0 +1,103 @@
+import org.apache.iotdb.db.metadata.tagSchemaRegion.utils.ConvertUtils;
+import org.apache.iotdb.lsm.annotation.FlushProcessor;
+import org.apache.iotdb.lsm.context.requestcontext.FlushRequestContext;
+import org.apache.iotdb.lsm.levelProcess.FlushLevelProcessor;
+import org.apache.iotdb.lsm.sstable.fileIO.FileOutput;
+
+import org.roaringbitmap.RoaringBitmap;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/** flush for MemChunk */
+@FlushProcessor(level = 2)
+public class MemChunkFlush extends FlushLevelProcessor<MemChunk, Object, FlushRequest> {
+
+  @Override
+  public List<Object> getChildren(
+      MemChunk memNode, FlushRequest request, FlushRequestContext context) {
+    return null;
+  }
+
+  @Override
+  public void flush(MemChunk memNode, FlushRequest request, FlushRequestContext context)
+      throws IOException {
+    FileOutput fileOutput = context.getFileOutput();
+    List<RoaringBitmap> roaringBitmaps = sliceMemChunk(memNode, request.getChunkMaxSize());
+    ChunkIndex chunkIndex = new ChunkIndex();
+    int count = 0;
+    for (RoaringBitmap roaringBitmap : roaringBitmaps) {
+      Chunk chunk = ConvertUtils.getChunkFromRoaringBitMap(roaringBitmap);
+      byte[] bytes = new byte[chunk.getChunkHeader().getSize()];
+      roaringBitmap.serialize(ByteBuffer.wrap(bytes));
+      fileOutput.write(bytes);
+      ChunkIndexEntry chunkIndexEntry =
+          ConvertUtils.getChunkIndexEntryFromRoaringBitMap(roaringBitmap);
+      chunkIndexEntry.setOffset(fileOutput.write(chunk.getChunkHeader()));
+      chunkIndex.getChunkIndexEntries().add(chunkIndexEntry);
+      count++;
+    }
+    chunkIndex.setChunkIndexHeader(new ChunkIndexHeader(count));
+    long offset = fileOutput.write(chunkIndex);
+    FlushResponse response = context.getResponse();
+    response.addChunkOffset(memNode, offset);
+  }
+
+  private List<RoaringBitmap> sliceMemChunk(MemChunk memNode, long chunkMaxSize) {
+    RoaringBitmap roaringBitmap = memNode.getRoaringBitmap();
+    int originalSize = roaringBitmap.serializedSizeInBytes();
+    int sliceNum = (int) (originalSize / chunkMaxSize) + 1;
+    if (sliceNum == 1) {
+      return Collections.singletonList(roaringBitmap);
+    } else {
+      List<RoaringBitmap> roaringBitmaps = new ArrayList<>();
+      for (int i = 0; i < sliceNum; i++) {
+        roaringBitmaps.add(new RoaringBitmap());
+      }
+      int[] results = roaringBitmap.stream().toArray();
+      int quotient = results.length % sliceNum;
+      int count = results.length / sliceNum;
+      int index = 0;
+      for (int i = 0; i < results.length; i++) {
+        int gap = index < quotient ? count + 1 : count;
+        int start = i;
+        for (; start <= i + gap; start++) {
+          roaringBitmaps.get(index).add(results[start]);
+        }
+        i = start;
+        index++;
+      }
+      return roaringBitmaps;
```

Comment:
It seems the last few roaringBitmaps may be empty.

Reply:
Add ut and is passed.